runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -8245,19 +8245,22 @@ class INTERPRETER_CLASS
 	{
 		VM_BytecodeAction rc = GOTO_RUN_METHOD;
 		bool fromJIT = J9_ARE_ANY_BITS_SET(_currentThread->jitStackFrameFlags, J9_SSF_JIT_NATIVE_TRANSITION_FRAME);
-		UDATA methodArgCount = 0;
+		UDATA mhReceiverIndex = 0;
 
 		if (fromJIT) {
-			methodArgCount = _currentThread->tempSlot;
+			/* tempSlot contains the number of stack slots for the arguments, and the MH
+			 * receiver is the first argument.
+			 */
+			mhReceiverIndex = _currentThread->tempSlot - 1;
```

Comment:
Would you consider adding some documentation for the `tempSlot` field? I personally get worried when we start overloading uses of a "temp" field. We are also increasingly growing what kinds of things we store into that field. Questions I ask myself when encountering a use of `tempSlot`:

- What are all the uses of this field?
- How can I be sure that the value I store here will not be overwritten by some other piece of code which wants to use this same field?

Please see #12045 where we attempt to document an overloaded field like this. I don't expect us to document all uses, but if you start a similar template as seen in #12045 for this particular use, then we can set a precedent that all future uses should be documented as well.

Eventually we may not eliminate such uses, but at the very least there will be concentrated documentation right where the field is declared about all the uses of it. Hopefully this will help the reader understand the answers to the above two questions.

Reply:
We've long discussed using individual fields for these things so that they could be somewhat self-documenting. The problem is that we need to keep the thread size down where possible, and that some of the fields that are accessed from the hand-coded assembly must be below a certain offset in order to use simple instruction forms on various processors. Similarly, some fields are accessed by the compiled code - those fields want to have small offsets to allow optimal encoding.

Reply:
@fjeremic, I agree that we should be documenting the usage of this field as much as we can. I have opened #12095 to track the work on documentation for this field.

Reply:
Thanks for opening that. I would like to see documentation added for the particular use case of this PR before we merge. Let's set the precedent and we can use the issue you opened to track addition of other documentation not related to this PR.